lib/bundler/ui/shell.rb

```diff
@@ -11,6 +11,7 @@ def initialize(options = {})
         end
         @shell = Thor::Base.shell.new
         @level = ENV['DEBUG'] ? "debug" : "info"
+        @told_messages = []
```

Comment:
@fotanus change the name of this ivar to `warning_history` or `past_warnings`, and this looks good to go.